src/provider/provider_factory.cpp

```
@@ -37,13 +45,70 @@ class MockProvider : public LLMProvider {
+      return match_scripted_response(script, prompt);
+    }
     const char* env = std::getenv("LLAMA_CLI_MOCK_RESPONSE");
     if (env) {
       return env;
     }
-    return "mock response: " + (messages.empty() ? "" : messages.back().content);
+    return "mock response: " + prompt;
+  }
+
+  /// Match response by keyword in prompt. Falls back to [default] block or echo.
+  static std::string match_scripted_response(const char* path, const std::string& prompt) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)